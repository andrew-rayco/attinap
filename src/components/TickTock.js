import React from 'react'
import PropTypes from 'prop-types'
import { Timepicker } from 'react-timepicker'

const TickTock = props => {
    return (
        <div className="time-picker">
            <button onClick={props.handleClockSubmit}>Submit time</button>
            <Timepicker
                hours={new Date().getHours()}
                minutes={new Date().getMinutes()}
                onChange={(hours, mins) => props.handleChange(hours, mins)}
                mode={props.timepickerMode}
            />
        </div>
    )
}

TickTock.propTypes = {
    handleChange: PropTypes.func.isRequired,
    handleClockSubmit: PropTypes.func.isRequired
}

export default TickTock
