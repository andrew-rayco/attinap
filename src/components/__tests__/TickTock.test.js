import React from 'react'
import ReactDOM from 'react-dom'
import { render } from '@testing-library/react'
import TickTock from '../TickTock'

test('Renders button with `submit time` text', () => {
    const container = document.createElement('div')
    ReactDOM.render(<TickTock />, container)
    expect(container.textContent).toMatch('Submit time')
})

// Only snapshotting button. No need to test Timepicker module
test('TickTock button snapshot', () => {
    const { getByText } = render(<TickTock />)
    const submitTimeButton = getByText('Submit time')

    expect(submitTimeButton).toMatchSnapshot()
})
