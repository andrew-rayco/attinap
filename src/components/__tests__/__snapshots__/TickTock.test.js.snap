// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TickTock button snapshot 1`] = `
<button>
  Submit time
</button>
`;
